### ISSUE TITLE
RTEMS: Error binding socket 10.0.2.15:5076: .

### ISSUE THREAD

**Issue by mdavidsaver** [epics-base/pvAccessCPP#75]
See QSRV and RTEMS/mvme3100 (4.9.6 and 4.10.2).

```
Error binding socket 10.0.2.15:5076: .
```

Enabling full debug information, and changing the code to print the numeric errno gives:

```
2017-11-27T18:43:22.568 Creating acceptor to 0.0.0.0:5075.
2017-11-27T18:43:22.588 Creating datagram socket to: 0.0.0.0:0.
2017-11-27T18:43:22.588 Starting thread: UDP-rx 0.0.0.0:0.
2017-11-27T18:43:22.588 Broadcast address #0: 10.0.2.255:5076.
2017-11-27T18:43:22.588 Setting up UDP for interface 10.0.2.15, broadcast 10.0.2.255, dest 10.0.2.255.
2017-11-27T18:43:22.588 Creating datagram socket to: 10.0.2.15:5076.
2017-11-27T18:43:22.588 Error binding socket 10.0.2.15:5076: (125) .
2017-11-27T18:43:22.588 Creating datagram socket to: 224.0.0.128:5076.
2017-11-27T18:43:22.588 Starting thread: UDP-rx 224.0.0.128:5076.
2017-11-27T18:43:22.588 Local multicast enabled on 127.0.0.1/224.0.0.128:5076.
2017-11-27T18:43:22.688 Accepting connections at 0.0.0.0:5075.
2017-11-27T18:43:22.708 Sending 47 bytes to 10.0.2.255:5076.
```

The UDP bind() is failing in blockingUDPConnector.cpp

**Comment by mdavidsaver** [comment]
With RTEMS/newlib, errno 125 is EADDRNOTAVAIL.

**Comment by mdavidsaver** [comment]
Fixed with 0161799c93bf17c45517eb64f2b5a7255ab32dd4.  Must zero sockaddr prior to use!

**Comment by mdavidsaver** [comment]
for future reference.  With at least RTEMS 4.9/4.10 the ```ifa_ifwithaddr()``` function is comparing sockaddr structs byte for byte.  See cpukit/libnetworking/net/if.c
